src/Module/BaseApi.php

```diff
@@ -101,6 +105,148 @@ public function run(ModuleHTTPException $httpException, array $request = [], boo
 		return parent::run($httpException, $request);
 	}
 
+	/**
+	 * Processes data from GET requests and sets paging conditions
+	 *
+	 * @param array $request       Custom REQUEST array
+	 * @param array $condition     Existing conditions to merge
+	 * @return array paging data condition parameters data
+	 * @throws \Exception
+	 */
+	protected function addPagingConditions(array $request, array $condition): array
+	{
+		$requested_order = $request['friendica_order'];
+		if ($requested_order == TimelineOrderByTypes::ID) {
+			if (!empty($request['max_id'])) {
+				$condition = DBA::mergeConditions($condition, ["`uri-id` < ?", intval($request['max_id'])]);
+			}
+
+			if (!empty($request['since_id'])) {
+				$condition = DBA::mergeConditions($condition, ["`uri-id` > ?", intval($request['since_id'])]);
+			}
+
+			if (!empty($request['min_id'])) {
+				$condition = DBA::mergeConditions($condition, ["`uri-id` > ?", intval($request['min_id'])]);
+			}
+		} else {
+			switch ($requested_order) {
+				case TimelineOrderByTypes::RECEIVED:
+				case TimelineOrderByTypes::CHANGED:
+				case TimelineOrderByTypes::EDITED:
+				case TimelineOrderByTypes::CREATED:
+				case TimelineOrderByTypes::COMMENTED:
+					$order_field = $requested_order;
+					break;
+				default:
+					throw new \Exception("Unrecognized request order: $requested_order");
+			}
+
+			if (!empty($request['max_id'])) {
+				$condition = DBA::mergeConditions($condition, ["`$order_field` < ?", DateTimeFormat::convert($request['max_id'], DateTimeFormat::MYSQL)]);
+			}
+
+			if (!empty($request['since_id'])) {
+				$condition = DBA::mergeConditions($condition, ["`$order_field` > ?", DateTimeFormat::convert($request['since_id'], DateTimeFormat::MYSQL)]);
+			}
+
+			if (!empty($request['min_id'])) {
+				$condition = DBA::mergeConditions($condition, ["`$order_field` > ?", DateTimeFormat::convert($request['min_id'], DateTimeFormat::MYSQL)]);
+			}
+		}
+
+		return $condition;
+	}
+
+	/**
+	 * Processes data from GET requests and sets paging conditions
+	 *
+	 * @param array $request  Custom REQUEST array
+	 * @param array $params   Existing $params element to build on
+	 * @return array ordering data added to the params blocks that was passed in
+	 * @throws \Exception
+	 */
+	protected function buildOrderAndLimitParams(array $request, array $params = []): array
+	{
+		$requested_order = $request['friendica_order'];
+		switch ($requested_order) {
+			case TimelineOrderByTypes::CHANGED:
+				$order_field = 'changed';
+				break;
+			case TimelineOrderByTypes::CREATED:
+				$order_field = 'created';
+				break;
+			case TimelineOrderByTypes::COMMENTED:
+				$order_field = 'commented';
+				break;
+			case TimelineOrderByTypes::EDITED:
+				$order_field = 'edited';
+				break;
+			case TimelineOrderByTypes::RECEIVED:
+				$order_field = 'received';
```

Comment:
You can pack these like you did in `addPagingConditions`
```suggestion
			case TimelineOrderByTypes::CHANGED:
			case TimelineOrderByTypes::CREATED:
			case TimelineOrderByTypes::COMMENTED:
			case TimelineOrderByTypes::EDITED:
			case TimelineOrderByTypes::RECEIVED:
				$order_field = $requested_order;
```